### DISCUSSION TITLE
Keep old name in minibuffer when renaming branch

### DISCUSSION THREAD

**Discussion by JimMoen** [magit/magit#4794] [Support] [answered]
The mini buffer input text is empty when renaming a branch.
But we just want to fix a typo in branch name in most cases.
Can we keep the old name when renaming branch. So that we can modify it easily.

**Comment by kyleam** [comment, marked as answer]
> Can we keep the old name when renaming branch. So that we can modify it easily.

When the branch name is read, it's added to `magit-revision-history`, which is then used when reading the new branch name, so you should be able to pull it in with `previous-history-element` (~~`M-n`~~ `M-p`).

https://github.com/magit/magit/blob/619b6a147cbadcd2e14c759b3e91e12f30acc79b/lisp/magit-branch.el#L718-L721

**Reply by JimMoen** [reply]
It's actually `M-p`.
